openeogeotrellis/deploy/batch_job.py

```diff
@@ -128,16 +135,47 @@ def extract_result_metadata(tracer: DryRunDataTracer) -> dict:
             # Intentionally don't return the complete vector file. https://github.com/Open-EO/openeo-api/issues/339
             geometry = mapping(Polygon.from_bounds(*bbox))
             area = agg_geometry.area
+            bbox_crs = agg_geometry.crs
         elif isinstance(agg_geometry, DriverVectorCube):
             bbox = agg_geometry.get_bounding_box()
             geometry = agg_geometry.get_bounding_box_geojson()
             area = agg_geometry.get_bounding_box_area()
+            bbox_crs = agg_geometry.get_crs()
         else:
             logger.warning(f"Result metadata: no bbox/area support for {type(agg_geometry)}")
 
+        # The aggregation geometries return tuples for their bounding box.
+        # Keep the end result consistent and convert it to a list.
+        if isinstance(bbox, tuple):
+            bbox = [*bbox]
+
     links = tracer.get_metadata_links()
     links = [link for k, v in links.items() for link in v]
 
+    # Convert bbox to lat-long, EPSG:4326 if it was any other CRS.
+    if bbox:
+        # Note that if the bbox comes from the aggregate_spatial_geometries, then we may
+        # get a pyproy CRS object instead of an EPSG code.That's fine. In that case we
+        # can simply go ahead with the reprojection.
+        # If the CRS happens to be EPSG:4326 after all then it will be a no-operation.
+        # It is difficult to list and check all possible variants of pyproj CRS objects
+        # that are actually all the exact same EPSG:4326 CRS, and it isn't really necessary.
+        if bbox_crs not in [4326, "EPSG:4326", "epsg:4326"]:
+            if aggregate_spatial_geometries:
+                new_spatial_extent = {
+                    "west": bbox[0],
+                    "south": bbox[1],
+                    "east": bbox[2],
+                    "north": bbox[3],
+                    "crs": bbox_crs,
+                }
+            else:
+                new_spatial_extent = spatial_extent
```

Comment:
I'm getting a warning that `spatial_extent` might not be initialized; you might want to account for the case where there is no extent.